### DISCUSSION TITLE
Mounting local disks on node

### DISCUSSION THREAD

**Discussion by stge4code** [ltsp/ltsp#398] [Q&A] [answered]
Dear developers,
Is it any restrictions to mount local disks on ltsp node?
I've connected to node (ltsp10)  via ssh and then successfully mount /dev/sda1 to /tmp/disk.
But  when I disconnect from the node and connect with it again, the disk becomes unmounted.

ltsp 21.01-1~ubuntu20.04.1

**Comment by alkisg** [comment]
Hi Alexey,

no, there's no such restriction. Although, maybe the following setting is related:

/etc/systemd/logind.conf: KillUserProcesses=yes

LTSP sets that to tell systemd to kill all the user processes on logout. Now I don't suppose that your mount command leaves any processes owned by the user behind, but you could check with `ps faux` before disconnecting from ssh.

**Reply by stge4code** [reply]
Thank you, Alkis!
I've checked it and found that the problem is connected with KillUserProcesses option.
Is it possible to switch off KillUserProcesses (set it to KillUserProcesses=no) for all nodes by default?

**Comment by alkisg** [comment, marked as answer]
Well, you could use something like this in ltsp.conf:

```shell
POST_INIT_LOGIND="sed 's/.*\(KillUserProcesses=\).*/\1no/' -i /etc/systemd/logind.conf"
```

But it's the WRONG thing to do, it will cause PROBLEMS when users log out if your display manager doesn't kill the processes itself. It would be best to mount thing properly and not as a child process. E.g. with fstab or with systemd mount units.

**Reply by stge4code** [reply]
Thanks again!

Should I reboot all my nodes after that or there is another way to apply ltsp.conf changes?

**Comment by alkisg** [comment]
The LTSP server doesn't have direct control over the clients, so `ltsp initrd` and client rebooting is needed to send new options.
But you can run that sed command with epoptes.org if you've installed it, and avoid rebooting.
